libraries/botbuilder-dialogs-adaptive/src/input/choiceInput.ts

```diff
@@ -130,15 +130,14 @@ export class ChoiceInput extends InputDialog implements ChoiceInputConfiguration
      * @param options Optional, initial information to pass to the dialog.
      * @returns The modified [ChoiceInputOptions](xref:botbuilder-dialogs-adaptive.ChoiceInputOptions) options.
      */
-    protected onInitializeOptions(dc: DialogContext, options: ChoiceInputOptions): ChoiceInputOptions {
+    protected async onInitializeOptions(dc: DialogContext, options: ChoiceInputOptions): Promise<ChoiceInputOptions> {
         if (!(options && options.choices && options.choices.length > 0)) {
             if (!options) {
                 options = { choices: [] };
             }
-            const choices = this.choices.getValue(dc.state);
-            options.choices = choices;
+            options.choices = await this.getChoiceSet(dc);
         }
-        return super.onInitializeOptions(dc, options);
+        return await super.onInitializeOptions(dc, options);
```

Comment:
`return await` is almost never needed, just return `super.onInitializeOptions` directly as it's already a Promise.

Reply:
But I should get the result of "this.getChoiceSet()" at first. So, there need a "await" to wait for the result.